src/mmg3d/colver_3d.c

```diff
@@ -393,7 +393,8 @@ int MMG3D_get_shellEdgeTag_oneDir(MMG5_pMesh  mesh,MMG5_int start, MMG5_int na,
     if ( pt->xt ) {
       pxt = &mesh->xtetra[pt->xt];
       *ref  = pxt->edg[i];
-      if ( pxt->tag[i] & MG_BDY ) {
+      if ((pxt->ftag[MMG5_ifar[i][0]] & MG_BDY) || (pxt->ftag[MMG5_ifar[i][1]] & MG_BDY)) {
+        *tag |= MG_BDY;
```

Comment:
- The `if ( pxt->tag[i] & MG_BDY )` can be kept as it may allow to exit the loop faster;
  - You can assign both `pxt->tag[i]` and `MG_BDY` tag using the bitewise OR operator `pxt->tag[i] | MG_BDY`
  
  Finally, the test must be something like that:
  ```C
  if ( pxt->tag[i] & MG_BDY ) {
    *tag |= pxt->tag[i];
    *filled =1;
     return adj;
  }  else if ((pxt->ftag[MMG5_ifar[i][0]] & MG_BDY) || (pxt->ftag[MMG5_ifar[i][1]] & MG_BDY)) {
    *tag |= (pxt->tag[i] | MG_BDY);
    *filled = 1;
     return adj;
  }
  ```